astroid/nodes/node_ng.py

```diff
@@ -248,15 +262,34 @@ def parent_of(self, node):
                 return True
         return False
 
-    def statement(self):
+    @overload
+    def statement(
+        self, future: Literal[None] = ...
+    ) -> Union["nodes.Statement", "nodes.Module"]:
+        ...
+
+    @overload
+    def statement(self, future: Literal[True]) -> "nodes.Statement":
+        ...
+
+    def statement(
+        self, future: Optional[Literal[True]] = None
```

Comment:
```suggestion
        self, future: Literal[None, True] = None
```
Might be a bit easier to understand, especially with the first overload.